packages/core/src/env-set/oidc.ts

```diff
@@ -1,66 +1,104 @@
  * @throws An error when failed to read a private key.
  */
-const readPrivateKey = async (): Promise<string> => {
-  const privateKey = getEnv('OIDC_PRIVATE_KEY');
+const readPrivateKeys = async (): Promise<string[]> => {
+  const privateKeys = getEnvAsStringArray('OIDC_PRIVATE_KEYS');
 
-  if (privateKey) {
-    return privateKey;
+  if (privateKeys.length > 0) {
+    return privateKeys;
   }
 
-  const privateKeyPath = getEnv('OIDC_PRIVATE_KEY_PATH', './oidc-private-key.pem');
+  // Downward compatibility for `OIDC_PRIVATE_KEY`
+  const compatPrivateKey = getEnv('OIDC_PRIVATE_KEY');
 
-  try {
-    return readFileSync(privateKeyPath, 'utf8');
-  } catch (error: unknown) {
-    if (noInquiry) {
-      throw error;
-    }
+  if (compatPrivateKey) {
+    return [compatPrivateKey];
+  }
 
-    if (!allYes) {
-      const answer = await inquirer.prompt({
-        type: 'confirm',
-        name: 'confirm',
-        message: `No private key found in env \`OIDC_PRIVATE_KEY\` nor \`${privateKeyPath}\`, would you like to generate a new one?`,
-      });
+  // Downward compatibility for `OIDC_PRIVATE_KEY_PATH`
+  const originPrivateKeyPath = getEnv('OIDC_PRIVATE_KEY_PATH');
+  const privateKeyPaths = getEnvAsStringArray(
+    'OIDC_PRIVATE_KEY_PATHS',
+    originPrivateKeyPath ? [originPrivateKeyPath] : []
+  );
 
-      if (!answer.confirm) {
+  if (privateKeyPaths.length === 0) {
+    try {
+      return [readFileSync(defaultLogtoOidcPrivateKey, 'utf8')];
+    } catch (error: unknown) {
+      if (noInquiry) {
         throw error;
       }
+
+      if (!allYes) {
+        const answer = await inquirer.prompt({
+          type: 'confirm',
+          name: 'confirm',
+          message: `No private key found in env \`OIDC_PRIVATE_KEYS\` nor \`${defaultLogtoOidcPrivateKey}\`, would you like to generate a new one?`,
+        });
+
+        if (!answer.confirm) {
+          throw error;
+        }
+      }
+
+      const { privateKey } = generateKeyPairSync('rsa', {
+        modulusLength: 4096,
+        publicKeyEncoding: {
+          type: 'spki',
+          format: 'pem',
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'pem',
+        },
+      });
+      writeFileSync(defaultLogtoOidcPrivateKey, privateKey);
+
+      return [privateKey];
     }
+  }
 
-    const { privateKey } = generateKeyPairSync('rsa', {
-      modulusLength: 4096,
-      publicKeyEncoding: {
-        type: 'spki',
-        format: 'pem',
-      },
-      privateKeyEncoding: {
-        type: 'pkcs8',
-        format: 'pem',
-      },
-    });
-    writeFileSync(privateKeyPath, privateKey);
+  const notExistPrivateKeys = privateKeyPaths.filter((path): boolean => !existsSync(path));
+
+  if (notExistPrivateKeys.length > 0) {
+    const notExistPrivateKeysRawValue = JSON.stringify(notExistPrivateKeys);
+    throw new Error(
+      `Private keys ${notExistPrivateKeysRawValue} configured in env \`OIDC_PRIVATE_KEY_PATHS\` not found.`
```

Comment:
trim the [] bracket?

Reply:
`notExistPrivateKeysRawValue` will be the same as the `OIDC_PRIVATE_KEY_PATHS` value.
It's ok to keep the `[]`, for this will not mislead the user if they should remove the `[]` or not in the config file.